include/cvc5/cvc5_skolem_id.h

```diff
@@ -0,0 +1,333 @@
+   */
+  EVALUE(BAGS_MAP_PREIMAGE),
+  EVALUE(BAGS_MAP_PREIMAGE_INJECTIVE),
+  /**
+   * A skolem variable for the size of the preimage of {y} that is unique per
+   * terms (bag.map f A), y which might be an element in (bag.map f A). (see the
+   * documentation for BAGS_MAP_PREIMAGE)
+   */
+  EVALUE(BAGS_MAP_PREIMAGE_SIZE),
+  /**
+   * A skolem variable for the index that is unique per terms
+   * (bag.map f A), y, preImageSize, y, e which might be an element in A.
+   * (see the documentation for BAGS_MAP_PREIMAGE)
+   */
+  EVALUE(BAGS_MAP_PREIMAGE_INDEX),
+  /** An uninterpreted function for bag.map operator:
+   * If the preimage of {y} in A is {uf(1), ..., uf(n)} (see BAGS_MAP_PREIMAGE},
+   * then the multiplicity of an element y in a bag (map f A) is sum(n),
+   * where sum: Int -> Int is a skolem function such that:
+   * sum(0) = 0
+   * sum(i) = sum (i-1) + (bag.count (uf i) A)
+   */
+  EVALUE(BAGS_MAP_SUM),
+  /** bag diff to witness (not (= A B)) */
+  EVALUE(BAGS_DEQ_DIFF),
+  /** Given a group term ((_ table.group n1 ... nk) A) of type (Bag (Table T))
+   * this uninterpreted function maps elements of A to their parts in the
+   * resulting partition. It has type (-> T (Table T))
+   */
+  EVALUE(TABLES_GROUP_PART),
+  /**
+   * Given a group term ((_ table.group n1 ... nk) A) of type (Bag (Table T))
+   * and a part B of type (Table T), this function returns a skolem element
+   * that is a member of B if B is not empty.
+   */
+  EVALUE(TABLES_GROUP_PART_ELEMENT),
+  /** Given a group term ((_ rel.group n1 ... nk) A) of type (Set (Relation T))
+   * this uninterpreted function maps elements of A to their parts in the
+   * resulting partition. It has type (-> T (Relation T))
+   */
+  EVALUE(RELATIONS_GROUP_PART),
+  /**
+   * Given a group term ((_ rel.group n1 ... nk) A) of type (Set (Relation T))
+   * and a part B of type (Relation T), this function returns a skolem element
+   * that is a member of B if B is not empty.
+   */
+  EVALUE(RELATIONS_GROUP_PART_ELEMENT),
+  /** An interpreted function for bag.choose operator:
+   * (choose A) is expanded as
+   * (witness ((x elementType))
+   *    (ite
+   *      (= A (as set.empty (Set E)))
+   *      (= x (uf A))
+   *      (and (set.member x A) (= x uf(A)))
+   * where uf: (Set E) -> E is a skolem function, and E is the type of elements
+   * of A
+   */
+  EVALUE(SETS_CHOOSE),
+  /** set diff to witness (not (= A B)) */
+  EVALUE(SETS_DEQ_DIFF),
+  EVALUE(SETS_FOLD_CARD),
+  EVALUE(SETS_FOLD_COMBINE),
+  EVALUE(SETS_FOLD_ELEMENTS),
+  EVALUE(SETS_FOLD_UNION),
+  /**
+   * A skolem variable that is unique per terms (set.map f A), y which is an
+   * element in (set.map f A). The skolem is constrained to be an element in A,
+   * and it is mapped to y by f.
+   */
+  EVALUE(SETS_MAP_DOWN_ELEMENT),
+  /** abstract value for a term t */
+  EVALUE(ABSTRACT_VALUE),
+  //================================================= Unknown rule
+  EVALUE(NONE),
```

Comment:
Maybe call it `LAST` for consistency...

Reply:
We actually need `NONE`, it is the expected return value when something is not a skolem and you ask its id.

We could add `LAST` as well, although I would do this separately when we determine what the overall policy for naming enums should be (note that `ProofRule` also does not use `LAST`).